samples/notebooks/prediction/SDK_Custom_Predict.ipynb

```diff
@@ -0,0 +1,1269 @@
+        "id": "212b2935ea12"
+      },
+      "source": [
+        "### Push the container to artifact registry\n",
+        "\n",
+        "Configure Docker to access Artifact Registry. Then push your container image to your Artifact Registry repository."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "wSFXCj3LdluJ"
+      },
+      "outputs": [],
+      "source": [
+        "!gcloud services list"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ABE9UpwSdluK"
+      },
+      "source": [
+        "If `artifactregistry.googleapis.com` is not enabled in your project, enable the API before proceeding."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "qDhLoQMydluK"
+      },
+      "outputs": [],
+      "source": [
+        "!gcloud services enable artifactregistry.googleapis.com"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "09ffe2434e3d"
+      },
+      "outputs": [],
+      "source": [
+        "!gcloud beta artifacts repositories create {REPOSITORY} \\\n",
+        "    --repository-format=docker \\\n",
+        "    --location=$REGION"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "293437024749"
+      },
+      "outputs": [],
+      "source": [
+        "!gcloud auth configure-docker {REGION}-docker.pkg.dev --quiet"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "1dd7448f4703"
+      },
+      "outputs": [],
+      "source": [
+        "!docker push {REGION}-docker.pkg.dev/{PROJECT_ID}/{REPOSITORY}/{IMAGE}"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "b438bfa2129f"
+      },
+      "source": [
+        "## Deploy to Vertex AI\n",
+        "\n",
+        "Use the Python SDK to upload and deploy your model."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "4ae19df6a33e"
+      },
+      "source": [
+        "### Upload the custom container model"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "8d682d8388ec"
+      },
+      "outputs": [],
+      "source": [
+        "from google.cloud import aiplatform"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "574fb82d3eed"
+      },
+      "outputs": [],
+      "source": [
+        "aiplatform.init(project=PROJECT_ID, location=REGION)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "2738154345d5"
+      },
+      "outputs": [],
+      "source": [
+        "endpoint.delete(force=True)\n",
```

Comment:
probably need to clean up this line and `model.delete()` line below?

Reply:
It's to call `delete` before `endpoint` and `model` are initialized if users follow the notebook from top to down. So there will be an exception since `endpoint` and `model` is `None`?

Reply:
Oops, yes, got confused which line this comment corresponded to :).